astropy/coordinates/representation.py

```diff
@@ -483,7 +554,141 @@ def represent_as(self, other_class):
+
+        # save the differentials, then strip them so that the copy doesn't do
+        # an unneeded copy, then re-attach them to the right objects
+        olddiffs = self._differentials
+        try:
+            self._differentials = None
+            newrepr = deepcopy(self)
+            newrepr._differentials = new_differentials
+            return newrepr
+        finally:
+            self._differentials = olddiffs
+
+    def without_differentials(self):
+        """
+        Create a new representation with the same positions as this
+        representation, but without any differentials.
+
+        Returns
+        -------
+        newrepr
+            A copy of this representation, but without the ``differentials``.
+        """
+
+        # Again, implemented this way for (1) performance (skips going through
+        # the initializer), and (2) will preserve any additional attributes a
+        # user may have set
+
+        # save the differentials, then strip them so that the copy doesn't do
+        # an unneeded copy, then re-attach them to the right objects
+        olddiffs = self._differentials
+        try:
+            self._differentials = None
+            newrepr = deepcopy(self)
+            newrepr._differentials = ()
+            return newrepr
+        finally:
+            self._differentials = olddiffs
+
+    def to_cartesian(self, diffstocart=False):
+        """
+        Convert the representation to its carteisan form.
+
+        Note that subclasses should *not* override this - rather they should
+        override the ``_to_cartesian_helper()`` method.
+
+        Parameters
+        ----------
+        diffstocart : bool
+          If True, any differentials this representation has are *also*
+          converted to their cartesian form.
+
+        Returns
+        -------
+        cartrepr : CartesianRepresentation
+          The representation in Cartesian form.
+        """
+        repr = self._to_cartesian_helper()
+        if diffstocart:
+            repr._differentials = tuple([
+                diff.represent_as(CartesianDifferential, base=self)
+                for diff in self._differentials])
+        else:
+            repr._differentials = self._differentials
+        return repr
+
+    @classmethod
+    def from_cartesian(cls, other):
+        """
+        Create a new representation in this class from a supplied
+        `CartesianRepresentation`.
+
+        Note that subclasses should *not* override this - rather they should
+        override the ``_from_cartesian_helper()`` method.
+
+        Parameters
+        ----------
+        other : BaseRepresentation object
+            The representation to turn into this class
+
+        Returns
+        -------
+        newrepr : object of this class
+            A new representation of this class's type.
+        """
+        repr = cls._from_cartesian_helper(other)
+        repr._differentials = other._differentials
```

Comment:
I think these need to be transformed here - if the user wants something more complicated, up to them to arrange that outside.